### DISCUSSION TITLE
What cropping method I should use if my foreground region is extremely small?

### DISCUSSION THREAD

**Discussion by NoTody** [Project-MONAI/MONAI#7154] [General]
Hi, I'm wondering what cropping method I should use in my case that foreground region is very small. I observe that if I use RandCropByPosNegLabel, many samples will have no foreground. I'm not sure about RandCropByLabelClasses. I think the cropped foreground should not always be centered because the sliding window inference crop would not always get foreground at the center?
If there are any tutorial on how to implement own version of Cropping transform? I would like to implement an improved version based on my special case.

**Comment by wyli** [comment]
that's a good point, to add some randomness to the foreground cropping locations, I think you can try `RandCropByPosNegLabel` with a large window followed by a `RandSpatialCrop` with a smaller window.
